packages/flutter_tools/test/integration.shard/android_obfuscate_test.dart

```diff
@@ -0,0 +1,113 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:typed_data';
+
+import 'package:archive/archive.dart';
+import 'package:file_testing/file_testing.dart';
+import 'package:flutter_tools/src/base/file_system.dart';
+
+import '../src/common.dart';
+import 'test_utils.dart';
+
+void main() {
+  late Directory tempDir;
+
+  setUp(() async {
+    tempDir = fileSystem.systemTempDirectory.createTempSync('flutter_tools_packages_test.');
+  });
+
+  tearDown(() {
+    tryToDelete(tempDir);
+  });
+
+  test('Dart identifiers are obfuscated with build apk --obfuscate', () async {
+    const String projectName = 'hello_world';
+    await processManager.run(<String>[
+      flutterBin,
+      'create',
+      '-t',
+      'module',
+      projectName,
+    ], workingDirectory: tempDir.path);
+    final String projectPath = tempDir.childDirectory(projectName).path;
+
+    await processManager.run(<String>[
+      flutterBin,
+      'build',
+      'apk',
+      '--target-platform=android-arm',
+      '--obfuscate',
+      '--split-debug-info=foo/',
+    ], workingDirectory: projectPath);
+
+    final File outputApkDirectory = fileSystem.file(
+      fileSystem.path.join(
+        projectPath,
+        'build',
+        'app',
+        'outputs',
+        'flutter-apk',
+        'app-release.apk',
+      ),
+    );
+
+    expect(outputApkDirectory, exists);
+    expect(_containsSymbol(outputApkDirectory, 'lib/armeabi-v7a/libapp.so', projectName), false);
+  });
+
+  test('Dart identifiers are obfuscated with build aar --obfuscate', () async {
+    const String moduleName = 'hello_module';
+    await processManager.run(<String>[
+      flutterBin,
+      'create',
+      '-t',
+      'module',
+      moduleName,
+    ], workingDirectory: tempDir.path);
+    final String projectPath = tempDir.childDirectory(moduleName).path;
+
+    await processManager.run(<String>[
+      flutterBin,
+      'build',
+      'aar',
+      '--target-platform=android-arm',
+      '--obfuscate',
+      '--split-debug-info=foo/',
+      '--no-debug',
+      '--no-profile',
+    ], workingDirectory: projectPath);
+
+    final File outputAarDirectory = fileSystem.file(
+      fileSystem.path.join(
+        projectPath,
+        'build',
+        'host',
+        'outputs',
+        'repo',
+        'com',
+        'example',
+        moduleName,
+        'flutter_release',
+        '1.0',
+        'flutter_release-1.0.aar',
+      ),
+    );
+
+    expect(outputAarDirectory, exists);
+    expect(_containsSymbol(outputAarDirectory, 'jni/armeabi-v7a/libapp.so', moduleName), false);
+  });
+}
+
+bool _containsSymbol(File outputArchive, String libappPath, String symbol) {
+  final Archive archive = ZipDecoder().decodeBytes(outputArchive.readAsBytesSync());
```

Comment:
A better version of https://github.com/flutter/flutter/blob/11bf180af0bd2b03ccbec636249250dbd10384f9/packages/flutter_tools/test/host_cross_arch.shard/ios_content_validation_test.dart#L278-L298
Inspired by 
https://github.com/flutter/flutter/blob/11bf180af0bd2b03ccbec636249250dbd10384f9/packages/flutter_tools/test/integration.shard/deferred_components_test.dart#L61-L65